packages/social-previews/src/helpers.tsx

```diff
@@ -86,6 +86,11 @@ export function preparePreviewText( text: string, options: PreviewTextOptions ):
 
 	let result = stripHtmlTags( text );
 
+	// Replace multiple new lines (2+) with 2 new lines
+	// There can be any whitespace characters in empty lines
+	// That is why "\s*"
+	result = result.replaceAll( /[\n\r]\s*[\n\r]+/g, '\n\n' );
```

Comment:
I know this is a small additional change, but should we do this as a separate PR?

I don't think this will match any leading whitespace characters on the first line.

Should we change it [to match 2 or more times](https://regex101.com/r/AcKaZy/1)?

```suggestion
	result = result.replaceAll( /(?:\s*[\n\r]){2,}/g, '\n\n' );
```

Reply:
That won't work for empty whitespace characters between the lines. We can fix that by using `.trim()` instead.